### ISSUE TITLE
Ref with anyOf not working as it should

### ISSUE THREAD

**Issue by davidgayoso** [sngular/scs-multiapi-plugin#202]
<img width="412" alt="Captura de pantalla 2023-04-17 a las 12 29 35" src="https://user-images.githubusercontent.com/126153821/232459179-61c06e77-5d15-4106-8f8c-df72c16572ef.png">
 
In the above case, I have a component with two refs to another component which contains an anyOf specification. The next photo shows the component with the anyOf: 

<img width="436" alt="Captura de pantalla 2023-04-17 a las 12 31 53" src="https://user-images.githubusercontent.com/126153821/232459580-9bf15cdb-93d0-4779-9827-08da4f72c732.png">

When the multiapi tries to generate all the needed files based on the OpenAPI specification, it returns a NullPointerException. However, if you substitute the first ref from the first image with the component itself like in the next image, it generates the corresponding elements but not properly:  
<img width="517" alt="Captura de pantalla 2023-04-17 a las 12 34 31" src="https://user-images.githubusercontent.com/126153821/232460195-61372dbe-0dee-4a61-8100-08ca1082bf37.png">

I think that the problem resides in how it works with the anyOf specification.

Here you have a zip containing the files with the OpenAPI specification:
[specification.zip](https://github.com/sngular/scs-multiapi-plugin/files/11249119/specification.zip)

**Comment by 5uso-sng** [comment]
Looking into this I managed to reproduce the NullPointerException with this reduced specification:
`api-test.yml`
```
openapi: 3.0.2
info:
  title: Schema Management
  version: 2.0.0
servers:
- url: https://localhost/v1
paths:
  /schema:
    get:
      tags:
      - schema
      operationId: getSchema
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: "assets/components.yml#/components/schemas/schema"
```
`assets/components.yml`
```
components:
  schemas:
    schema:
      type: object
      properties:
        thing:
          $ref: "#/components/schemas/typeArray"
    typeArray:
      type: array
      items:
        oneOf:
        - $ref: "#/components/schemas/stringField"
    stringField:
      type: string
```

This crashes inside the MapperContentUtil class. If `schema` is instead defined as...

```
schema:
  $ref: "#/components/schemas/typeArray"
```

...the crash then happens inside the MapperPathUtilCass.

Do note that separating the components into a different file is required for the issue to occur. This seems to be caused  by the OpenAPI library returning an incomplete list of schemas when calling `openAPI.getComponents().getSchemas()` only when the files are separate.

**Comment by AdrianLagarteraSngular** [comment]
Resolved in the #160
